src/collection-view.js

```diff
@@ -457,9 +460,13 @@ Marionette.CollectionView = Marionette.View.extend({
     return view;
   },
 
-  // check if the collection is empty
-  isEmpty: function() {
-    return !this.collection || this.collection.length === 0;
+  // check if the collection is empty, taking the filter into account unless 'skipFilter' is true
+  isEmpty: function(collection, options) {
```

Comment:
how about a variant of option 1...

Only reference the collection if its passed in else  use `this.collection`. I think this would maintain BC?